src/config.rs

```diff
@@ -269,23 +269,59 @@ pub struct WindowConfig {
     dimensions: Dimensions,
 
     /// Pixel padding
-    #[serde(default="default_padding", deserialize_with = "deserialize_padding")]
-    padding: Delta<u8>,
+    #[serde(default, deserialize_with = "deserialize_padding")]
+    padding: Padding,
 
     /// Draw the window with title bar / borders
     #[serde(default, deserialize_with = "failure_default")]
     decorations: bool,
 }
 
-fn default_padding() -> Delta<u8> {
-    Delta { x: 2, y: 2 }
+fn default_padding() -> Padding {
+    Padding {
+        top: default_padding_field(),
+        right: default_padding_field(),
+        bottom: default_padding_field(),
+        left: default_padding_field(),
+        x: default_padding_field_deprecated(),
+        y: default_padding_field_deprecated(),
+    }
+}
+
+fn default_padding_field() -> u8 {
```

Comment:
I think that setting `x/y` to just `None` would be perfectly fine then there wouldn't be the scenario where `x/y` is removed because of deprecation but the method still stays around (linter should warn though). But I think this is totally fine the way it is too.

Reply:
Also there's no reason to have a `default_padding` method. `impl Default for Padding` seems like a better choice.
This would also allow using `..Default::default()` for `x/y`.

Reply:
I was just trying to work with the existing `default_padding` function, but you're right that's much better.